Tests/CreateDumpTest.php

```diff
@@ -0,0 +1,195 @@
+    public function createDestinationFilePathWithSubFolder()
+    {
+        Config::set('protector.baseDirectory', 'noDumps');
+
+        $this->disk->deleteDirectory(Config::get('protector.baseDirectory'));
+
+        $filePath            = $this->protector->createDestinationFilePath(__FUNCTION__, __FUNCTION__);
+        $method              = $this->getAccessibleReflectionMethod('createDirectory');
+        $destinationFilePath = $this->disk->path($filePath);
+
+        $method->invoke($this->protector, $destinationFilePath);
+
+        $this->assertDirectoryExists($destinationFilePath);
+    }
+
+    /**
+     * @test
+     */
+    public function failDirectoryCreationOnInvalidPath()
+    {
+        $method = $this->getAccessibleReflectionMethod('createDirectory');
+        $path   = 'https://example.com/protector/exportDump';
+
+        $this->expectException(FailedCreatingDestinationPathException::class);
+        $method->invoke($this->protector, $path);
+    }
+
+
+    /**
+     * @test
+     */
+    public function createMetaData()
+    {
+        $method = $this->getAccessibleReflectionMethod('getMetaData');
+
+        $method->invoke($this->protector);
+        $metaData = $method->invoke($this->protector, false);
+
+        $this->assertIsArray($metaData);
+    }
+
+    /**
+     * @test
+     */
+    public function verifyDumpDateMetaData()
+    {
+        $dumpMetaData = $this->protector->getDumpMetaData($this->filePath);
+
+        $date   = $dumpMetaData['meta']['dumpedAtDate'];
+        $result = checkDate($date['mon'], $date['wday'], $date['year']);
+
+        foreach (array_keys($date) as $key)
+        {
+            if (!isset($date[$key]))
+            {
+                $result = false;
+            }
+        }
+
+        $this->assertTrue($result);
+        $this->assertIsArray($dumpMetaData);
+    }
+
+    /**
+     * @test
+     * @define-env usesEmptyDump
+     */
+    public function failOnDumpHasNoMetaData()
+    {
+        $metaData = $this->protector->getDumpMetaData($this->emptyDumpPath);
+
+        $this->assertFalse($metaData);
+    }
+
+    /**
+     * @test
+     * @define-env usesEmptyDump
+     */
+    public function failOnDumpHasIncorrectMetaData()
+    {
+        $this->disk->put($this->emptyDumpPath, sprintf("%s\n%s", __FUNCTION__, __FUNCTION__));
+
+        $metaData = sprintf("-- options:%s\n-- meta:%s", __FUNCTION__, __FUNCTION__);
+        $this->disk->append($this->emptyDumpPath, $metaData);
+
+        $result = $this->protector->getDumpMetaData($this->emptyDumpPath);
+
+        $this->assertFalse($result);
+    }
+
+    /**
+     * @test
+     * @define-env usesEmptyDump
+     */
+    public function canGenerateDump()
+    {
+        $this->disk->put($this->emptyDumpPath, __FUNCTION__);
+
+        $method = $this->getAccessibleReflectionMethod('generateDump');
+        $result = $method->invoke($this->protector, $this->emptyDumpPath, ['no-data' => true]);
+
+        $this->assertTrue($result);
```

Comment:
taking into consideration that we currently have no database at this point, it is alarming that this would be true

Reply:
Fixed.